src/flamenco/runtime/fd_blockstore.c

```diff
@@ -33,59 +33,78 @@ fd_blockstore_align( void ) {
 }
 
 ulong
-fd_blockstore_footprint( void ) {
-  return sizeof( fd_blockstore_t );
+fd_blockstore_footprint( ulong shred_max, ulong block_max, ulong txn_max, ulong alloc_max ) {
+  ulong l = FD_LAYOUT_FINI(
```

Comment:
You are using fd_wksp_alloc_laddr to allocate the parts, which is inconsistent with just laying out the pieces consecutively